src/m1_facilitator/manifest/Manifest.ts

```diff
@@ -0,0 +1,304 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+import { Validator as JsonSchemaVerifier } from 'jsonschema';
+import { EncryptedKeystoreV3Json } from 'web3-eth-accounts';
+import schema from './manifest.schema.json';
+import Web3 from 'web3';
+
+/**
+ * Interface of facilitator manifest file input chain data. It represents below:
+ * {
+ *    avatar_account: ""
+ *    node_endpoint: ""
+ *    graph_ws_endpoint: ""
+ *    graph_rpc_endpoint: ""
+ * }
+ */
+interface ManifestChain {
+  avatar_account: string;
+  node_endpoint: string;
+  graph_ws_endpoint: string;
+  graph_rpc_endpoint: string;
+}
+
+/**
+ * Interface of facilitator manifest file input account data. It represents below:
+ * {
+ *    keystore_path: ""
+ *    keystore_password_path: ""
+ * }
+ */
+interface ManifestAccount {
+  keystore_path: string;
+  keystore_password_path: string;
+}
+
+/**
+ * Interface of facilitator manifest input file. It represents below:
+ * {
+ *    version: ""
+ *    architecture_layout: ""
+ *    personas: []
+ *    metachain:
+ *      origin: object
+ *      auxiliary: object
+ *    accounts: object
+ *    origin_contract_addresses: object
+ *    facilitate_tokens: []
+ * }
+ */
+interface ManifestInfo {
+  version: string;
+  architecture_layout: string;
+  personas: string[];
+  metachain: {
+    origin: ManifestChain;
+    auxiliary: ManifestChain;
+  };
+  accounts: Record<string, ManifestAccount>;
+  origin_contract_addresses: Record<string, string>;
+  facilitate_tokens: string[];
+}
+
+/** Enum of architecture layouts which facilitator supports. */
+enum ArchitectureLayout {
+  MOSAIC1 = 'M1',
```

Comment:
```suggestion
  MOSAIC1 = 'MOSAIC1',
```
In manifest.schema.json its `MOSAIC1`